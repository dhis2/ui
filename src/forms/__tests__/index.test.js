/* eslint-disable import/namespace */

import { FinalForm, ReactFinalForm } from '../index.js'

describe('final-form named export', () => {
    it('should re-export the expected exports', () => {
        const exports = Object.keys(FinalForm)

        expect(exports).toMatchSnapshot()
    })

    it.each(Object.keys(FinalForm))(
        'should re-export the expected type for %s',
        (exportName) => {
            const type = typeof FinalForm[exportName]

            expect(type).toMatchSnapshot()
        }
    )
})

describe('react-final-form named export', () => {
    it('should re-export the expected exports', () => {
        const exports = Object.keys(ReactFinalForm)

        expect(exports).toMatchSnapshot()
    })

    it.each(Object.keys(ReactFinalForm))(
        'should re-export the expected type for %s',
        (exportName) => {
            const type = typeof ReactFinalForm[exportName]

            expect(type).toMatchSnapshot()
        }
    )
})
