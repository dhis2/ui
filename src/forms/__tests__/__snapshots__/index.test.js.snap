// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`final-form named export should re-export the expected exports 1`] = `
Array [
  "ARRAY_ERROR",
  "FORM_ERROR",
  "configOptions",
  "createForm",
  "fieldSubscriptionItems",
  "formSubscriptionItems",
  "getIn",
  "setIn",
  "version",
  "default",
]
`;

exports[`final-form named export should re-export the expected type for ARRAY_ERROR 1`] = `"string"`;

exports[`final-form named export should re-export the expected type for FORM_ERROR 1`] = `"string"`;

exports[`final-form named export should re-export the expected type for configOptions 1`] = `"object"`;

exports[`final-form named export should re-export the expected type for createForm 1`] = `"function"`;

exports[`final-form named export should re-export the expected type for default 1`] = `"object"`;

exports[`final-form named export should re-export the expected type for fieldSubscriptionItems 1`] = `"object"`;

exports[`final-form named export should re-export the expected type for formSubscriptionItems 1`] = `"object"`;

exports[`final-form named export should re-export the expected type for getIn 1`] = `"function"`;

exports[`final-form named export should re-export the expected type for setIn 1`] = `"function"`;

exports[`final-form named export should re-export the expected type for version 1`] = `"string"`;

exports[`react-final-form named export should re-export the expected exports 1`] = `
Array [
  "Field",
  "Form",
  "FormSpy",
  "useField",
  "useForm",
  "useFormState",
  "version",
  "withTypes",
]
`;

exports[`react-final-form named export should re-export the expected type for Field 1`] = `"object"`;

exports[`react-final-form named export should re-export the expected type for Form 1`] = `"function"`;

exports[`react-final-form named export should re-export the expected type for FormSpy 1`] = `"function"`;

exports[`react-final-form named export should re-export the expected type for useField 1`] = `"function"`;

exports[`react-final-form named export should re-export the expected type for useForm 1`] = `"function"`;

exports[`react-final-form named export should re-export the expected type for useFormState 1`] = `"function"`;

exports[`react-final-form named export should re-export the expected type for version 1`] = `"string"`;

exports[`react-final-form named export should re-export the expected type for withTypes 1`] = `"function"`;
